src/main/java/com/atlassian/bitbucket/jenkins/internal/trigger/BitbucketWebhookConsumer.java

```diff
@@ -69,9 +69,20 @@ void process(MirrorSynchronizedWebhookEvent event) {
 
     void process(PullRequestWebhookEvent event) {
         LOGGER.fine(format("Received pull request event"));
-        // TODO: Can we do eligible refs? For now building everything
         RefChangedDetails refChangedDetails = new RefChangedDetails(event);
-        triggerJob(event, refChangedDetails);
+        Optional<BitbucketServerConfiguration> server = bitbucketPluginConfiguration.getValidServerList()
+                .stream()
+                .filter(serverConfig -> refChangedDetails.getRepository()
+                        .getSelfLink()
+                        .contains(serverConfig.getBaseUrl()))
+                .findFirst();
+        if (event.getEventKey().equals(BitbucketWebhookEvent.PULL_REQUEST_OPENED_EVENT.getEventId())) {
+            triggerJob(event, refChangedDetails, server);
+        } else {
```

Comment:
Having the various closed events in a single enum would also mean you could make this else loop explicit